### DISCUSSION TITLE
Use of ZOWE CLI with RSE API plugin to execute DBB scripts on USS

### DISCUSSION THREAD

**Discussion by wfezzani** [IBM/dbb#204] [Q&A]
Hello
It seems that  when ZOWE CLI with RSE API to run USS commands, it doesn't retrieve the PATH environment variable of the user who submits the command. ZOWE CLI seems apparently to use another user BPXROOT instead of the TSO user and it inherits the PATH env variable of that user. Moreover, theoretically the .profile file of the BPXROOT user (if it exists because I can't find it) is not supposed to be modified to edit the PATH env variable.
This can be a real limitation when we want  to execute the common backend scripts like gitClone.sh if the PATH variable doesn't include the git program.
In my opinion the ZOWE framework doesn't include yet the new RSE API: /rseapi/api/v1/unixcommands/shellcommand
and still uses /the API: rseapi/api/v1/unixcommands
To make it work we need to source the .profile content in every Zowe command.
Here is an exemple for git command:
```
[root@srv210-3202 ~]# zowe rse issue uss "git" --cwd .
git: FSUM7351 not found
[root@srv210-3202 ~]# 
```
and for whoami command:

```
[root@srv210-3202 ~]# zowe rse issue uss "whoami" --cwd .
BPXROOT
[root@srv210-3202 ~]# 
```

**Comment by dennis-behm** [comment]
@wfezzani - if your used id has UID=0, then you are BPXROOT. No way around this ;) You can check if the home directory is set correctly.

RSE API does not automatically run through your personal .profile, because it is not opening a SSH session.

As indicated in the the [rse configuration](https://www.ibm.com/docs/en/explorer-for-zos/3.3?topic=bc-rseenv-rse-configuration-file), you can configure and extend PATH and add your git directories.

```
# Specify additional command search directories
#-------------------------------------------------------------
#PATH=$PATH:/bin
```

Additionally, you may want to check the [RSEAPI configuration](https://www.ibm.com/docs/en/explorer-for-zos/3.3?topic=customization-rseapienv-rse-api-configuration-file). 


Alternatively, you can always execute your profile first:
```
zowe rse issue unix ". /u/ibmuser/.profile && git --version"
```

How this helps you to resolve it. To ask for the RSE API plugin, you might want to raise a question in the [Slack channel of ZOWE CLI](https://openmainframeproject.slack.com/archives/CC8AALGN6) in the openmainframeproject.

**Comment by wfezzani** [comment]
Hi Dennis, thank you for your feed back 
I have already tried the solutions you suggested to edit the PATH variable in both rse.env and rseapi.env but it didn't resolve the problem.
I think the problem comes from Zowe itself and not from RSE/RSE API.
I will raise the problem within the ZOWE community.
Thank you again.

**Reply by dennis-behm** [reply]
@wfezzani I guess it is rather an discussion topic for the RSEAPI team, if the documented functionality does not work as we expect it.
